From: Ondrej Holy <oholy@redhat.com>
Date: Wed, 7 Feb 2018 16:26:49 +0100
Subject: user-accounts: Remove unused cell renderer

With the carousel, the custom user image cell renderer is no more needed.
---
 panels/user-accounts/meson.build                   |   1 -
 panels/user-accounts/um-cell-renderer-user-image.c | 137 ---------------------
 panels/user-accounts/um-cell-renderer-user-image.h |  53 --------
 panels/user-accounts/um-user-panel.c               |   2 -
 4 files changed, 193 deletions(-)
 delete mode 100644 panels/user-accounts/um-cell-renderer-user-image.c
 delete mode 100644 panels/user-accounts/um-cell-renderer-user-image.h

diff --git a/panels/user-accounts/meson.build b/panels/user-accounts/meson.build
index 038e8bc..c2b73e3 100644
--- a/panels/user-accounts/meson.build
+++ b/panels/user-accounts/meson.build
@@ -148,7 +148,6 @@ sources = common_sources + files(
   'run-passwd.c',
   'um-account-type.c',
   'um-carousel.c',
-  'um-cell-renderer-user-image.c',
   'um-fingerprint-dialog.c',
   'um-history-dialog.c',
   'um-password-dialog.c',
diff --git a/panels/user-accounts/um-cell-renderer-user-image.c b/panels/user-accounts/um-cell-renderer-user-image.c
deleted file mode 100644
index bcd6160..0000000
--- a/panels/user-accounts/um-cell-renderer-user-image.c
+++ /dev/null
@@ -1,137 +0,0 @@
-/*
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
- *
- * (C) Copyright 2015 Red Hat, Inc.
- */
-
-#include "um-cell-renderer-user-image.h"
-
-#include <gtk/gtk.h>
-#include <act/act.h>
-
-#include "um-utils.h"
-
-struct _UmCellRendererUserImagePrivate {
-        GtkWidget *parent;
-        ActUser *user;
-};
-
-#define UM_CELL_RENDERER_USER_IMAGE_GET_PRIVATE(obj) (G_TYPE_INSTANCE_GET_PRIVATE ((obj), UM_TYPE_CELL_RENDERER_USER_IMAGE, UmCellRendererUserImagePrivate))
-
-enum {
-        PROP_0,
-        PROP_PARENT,
-        PROP_USER
-};
-
-G_DEFINE_TYPE_WITH_CODE (UmCellRendererUserImage, um_cell_renderer_user_image, GTK_TYPE_CELL_RENDERER_PIXBUF, G_ADD_PRIVATE (UmCellRendererUserImage));
-
-static void
-render_user_image (UmCellRendererUserImage *cell_renderer)
-{
-        cairo_surface_t *surface;
-        gint scale;
-
-        if (cell_renderer->priv->user != NULL) {
-                scale = gtk_widget_get_scale_factor (cell_renderer->priv->parent);
-                surface = render_user_icon (cell_renderer->priv->user, UM_ICON_STYLE_FRAME | UM_ICON_STYLE_STATUS, 48, scale);
-                g_object_set (GTK_CELL_RENDERER_PIXBUF (cell_renderer), "surface", surface, NULL);
-                cairo_surface_destroy (surface);
-        } else {
-                g_object_set (GTK_CELL_RENDERER_PIXBUF (cell_renderer), "surface", NULL, NULL);
-        }
-}
-
-static void
-on_scale_factor_changed (GObject    *object,
-                         GParamSpec *pspec,
-                         gpointer    data)
-{
-        UmCellRendererUserImage *cell_renderer = UM_CELL_RENDERER_USER_IMAGE (data);
-
-        render_user_image (cell_renderer);
-}
-
-static void
-um_cell_renderer_user_image_set_property (GObject      *object,
-                                          guint         prop_id,
-                                          const GValue *value,
-                                          GParamSpec   *pspec)
-{
-        UmCellRendererUserImage *cell_renderer = UM_CELL_RENDERER_USER_IMAGE (object);
-
-        switch (prop_id) {
-        case PROP_PARENT:
-                cell_renderer->priv->parent = g_value_dup_object (value);
-                g_signal_connect (cell_renderer->priv->parent, "notify::scale-factor", G_CALLBACK (on_scale_factor_changed), cell_renderer);
-                break;
-        case PROP_USER:
-                g_clear_object (&cell_renderer->priv->user);
-                cell_renderer->priv->user = g_value_dup_object (value);
-                render_user_image (cell_renderer);
-                break;
-        default:
-                G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
-                break;
-        }
-}
-
-static void
-um_cell_renderer_user_image_finalize (GObject *object)
-{
-        UmCellRendererUserImage *cell_renderer = UM_CELL_RENDERER_USER_IMAGE (object);
-
-        g_clear_object (&cell_renderer->priv->parent);
-        g_clear_object (&cell_renderer->priv->user);
-
-        G_OBJECT_CLASS (um_cell_renderer_user_image_parent_class)->finalize (object);
-}
-
-static void
-um_cell_renderer_user_image_class_init (UmCellRendererUserImageClass *class)
-{
-        GObjectClass *object_class;
-
-        object_class = G_OBJECT_CLASS (class);
-
-        object_class->set_property = um_cell_renderer_user_image_set_property;
-        object_class->finalize = um_cell_renderer_user_image_finalize;
-
-        g_object_class_install_property (object_class, PROP_PARENT,
-                                         g_param_spec_object ("parent",
-                                                              "Parent",
-                                                              "Tree view aprent widget",
-                                                              GTK_TYPE_WIDGET,
-                                                              G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY));
-
-        g_object_class_install_property (object_class, PROP_USER,
-                                         g_param_spec_object ("user",
-                                                              "User",
-                                                              "Accountsservice user used to generate image",
-                                                              ACT_TYPE_USER,
-                                                              G_PARAM_WRITABLE));
-}
-
-static void
-um_cell_renderer_user_image_init (UmCellRendererUserImage *cell_renderer)
-{
-        cell_renderer->priv = UM_CELL_RENDERER_USER_IMAGE_GET_PRIVATE (cell_renderer);
-}
-
-GtkCellRenderer *
-um_cell_renderer_user_image_new (GtkWidget *parent)
-{
-        return g_object_new (UM_TYPE_CELL_RENDERER_USER_IMAGE, "parent", parent, NULL);
-}
diff --git a/panels/user-accounts/um-cell-renderer-user-image.h b/panels/user-accounts/um-cell-renderer-user-image.h
deleted file mode 100644
index 6176d2b..0000000
--- a/panels/user-accounts/um-cell-renderer-user-image.h
+++ /dev/null
@@ -1,53 +0,0 @@
-/*
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
- *
- * (C) Copyright 2015 Red Hat, Inc.
- */
-
-#ifndef _UM_CELL_RENDERER_USER_IMAGE_H
-#define _UM_CELL_RENDERER_USER_IMAGE_H
-
-#include <gtk/gtk.h>
-
-G_BEGIN_DECLS
-
-#define UM_TYPE_CELL_RENDERER_USER_IMAGE um_cell_renderer_user_image_get_type()
-
-#define UM_CELL_RENDERER_USER_IMAGE(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), UM_TYPE_CELL_RENDERER_USER_IMAGE, UmCellRendererUserImage))
-#define UM_CELL_RENDERER_USER_IMAGE_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST ((klass), UM_TYPE_CELL_RENDERER_USER_IMAGE, UmCellRendererUserImageClass))
-#define UM_IS_CELL_RENDERER_USER_IMAGE(obj) (G_TYPE_CHECK_INSTANCE_TYPE ((obj), UM_TYPE_CELL_RENDERER_USER_IMAGE))
-#define UM_IS_CELL_RENDERER_USER_IMAGE_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), UM_TYPE_CELL_RENDERER_USER_IMAGE))
-#define UM_CELL_RENDERER_USER_IMAGE_GET_CLASS(obj) (G_TYPE_INSTANCE_GET_CLASS ((obj), UM_TYPE_CELL_RENDERER_USER_IMAGE, UmCellRendererUserImageClass))
-
-typedef struct _UmCellRendererUserImage UmCellRendererUserImage;
-typedef struct _UmCellRendererUserImageClass UmCellRendererUserImageClass;
-typedef struct _UmCellRendererUserImagePrivate UmCellRendererUserImagePrivate;
-
-struct _UmCellRendererUserImage {
-        GtkCellRendererPixbuf parent;
-
-        UmCellRendererUserImagePrivate *priv;
-};
-
-struct _UmCellRendererUserImageClass {
-        GtkCellRendererPixbufClass parent_class;
-};
-
-GType            um_cell_renderer_user_image_get_type (void) G_GNUC_CONST;
-GtkCellRenderer *um_cell_renderer_user_image_new      (GtkWidget *parent);
-
-G_END_DECLS
-
-#endif /* _UM_CELL_RENDERER_USER_IMAGE_H */
diff --git a/panels/user-accounts/um-user-panel.c b/panels/user-accounts/um-user-panel.c
index b43bbfd..15aeb29 100644
--- a/panels/user-accounts/um-user-panel.c
+++ b/panels/user-accounts/um-user-panel.c
@@ -39,7 +39,6 @@
 #include <libgnome-desktop/gnome-languages.h>
 
 #include "um-user-image.h"
-#include "um-cell-renderer-user-image.h"
 
 #include "um-account-dialog.h"
 #include "cc-language-chooser.h"
@@ -1444,7 +1443,6 @@ cc_user_panel_init (CcUserPanel *self)
 
         /* register types that the builder might need */
         type = um_user_image_get_type ();
-        type = um_cell_renderer_user_image_get_type ();
         type = um_carousel_get_type ();
 
         d->builder = gtk_builder_new ();
